_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+
+To sum up:
+* Create architecture diagrams and review them
+* Threat modeling 
+* Evil user stories
+
+
+#### Implementation
+
+Implementation is where the stuff that developers love happens. There are numerous ways to improve security or fail bad at this phase. Most important thing will be your developers security awareness. Many of the security issues are more or less stuff that nobody came to think about. Yes, I mean the OWASP Top 10 stuff like SQL injections. One thing you need to understand about developers is that they love to solve problems. So normal trick is to try to dress security requirements as a solvable problems like "make sure that all input is sanitized against sql injections" and "make sure that all output is sanitized against XSS". This is important but of course covers only one problem at a time and it will become a pretty exhaustive list to implement all over again in every project. Often times more eye-pairs for the same problem will help in finding loopholes in your security design. 
+
+As a software developer and security enthusiastic I could go on forever on this topic but more or less I will just put a list here things that you should consider:
+* Know or setup secure coding practices for your language and frameworks
+* Validate all the 3rd party libraries you take in
+* Fail securely
+* Make sure that your secrets management endures the daylight
+* Know the http security headers or atleast know how to check them
+* Understand the cryptography you are using
+* Basically memorize the whole OWASP TOP 10 and constantly consider them as a threats
+* Use XP programming
+* Have a review process
+* Find tooling that supports your work
+* Sanitize inputs
+* Sanitize outputs
+
+#### Test
+
+Security testing is one that works relatively same as every testing. You should just remember that now that we have security requirements we will also need to verify that we fulfill them. Some helpful things to do is to setup a good continuous integration environment that gives your development team constant feedback about how they are doing. If they are more on the DevOps or DevSecOps then they most likely will build it by themselves but some organization have different people for these tasks. I consider continuous integration as a testing as it is something that is outsourced for a system that gives you a feedback about your work. Some could also think that it is part of the implementation. 
```

Comment:
"Security testing is one that works relatively same as every testing" parempi muoto esim. "Security testing works relatively similar to any other kind of testing"

Reply:
"You should just remember that now that we have security requirements we will also need to verify that we fulfill them" -> "You should remember that now that we have security requirements we also need to verify that we fulfill them"

Reply:
Some helpful things -> A helpful thing

Reply:
"If they are more on the DevOps or DevSecOps then they most likely will build it by themselves but some organization have different people for these tasks." -> "If they are more into DevOps or DevSecOps then they most likely will build it by themselves but some organizations have different people for these tasks."
"I consider continuous integration as a testing" a pois.
"outsourced for a system that gives you a feedback about your work" -> "outsourced to a system that gives you feedback on your work"